packages/fxa-settings/src/components/Ready/index.tsx

```diff
@@ -4,14 +4,44 @@
 
 import React from 'react';
 import { RouteComponentProps } from '@reach/router';
-import { Localized } from '@fluent/react';
+import { FtlMsg } from 'fxa-react/lib/utils';
 import { ReactComponent as PulseHearts } from './account-verified.svg';
 import { logViewEvent, usePageViewEvent } from '../../lib/metrics';
 
 type ReadyProps = {
   serviceName?: string;
   continueHandler?: Function;
-  viewName: string;
+  viewName: viewNameType;
+};
+
+type viewNameType =
+  | 'signin-confirmed'
+  | 'signin-verified'
+  | 'reset-password-confirmed'
+  | 'reset-password-verified'
+  | 'reset-password-with-recovery-key-verified';
+
+const getTemplateValues = (viewName: viewNameType) => {
+  let templateValues: { headerText: string; headerId: string } = {
+    headerText: '',
+    headerId: '',
+  };
+  // I use the switch case here instead of just a hash because there's a lot of duplication,
+  // and I think the fallthrough structure is really intuitive! Open to feedback on this --
+  // if it seems unclear to people, that's a good reason to change it
+  switch (viewName) {
+    case 'signin-confirmed':
+    case 'signin-verified':
+      templateValues.headerId = 'fxa-sign-in-complete-header';
+      templateValues.headerText = 'Sign-in confirmed';
+      break;
+    case 'reset-password-confirmed':
+    case 'reset-password-with-recovery-key-verified':
+      templateValues.headerId = 'fxa-reset-password-complete-header';
+      templateValues.headerText = 'Your password has been reset';
+      break;
+  }
```

Comment:
Hhmm. Should this have a `default` case which throws some form of error?
What happens if `viewName` mysteriously isn't one of these four cases, the `headerId`/`headerText` are empty strings; but not sure what downstream issues that might cause.

Reply:
Oh yeah, thanks Peter. I meant to call this out as well.

Reply:
I get this instinct, but I think omitting the default case is actually clearer -- the viewName has to be one of these cases due to the `ViewNameType` requirement, so adding an impossible default case seems misleading

Reply:
@millsoper You could potentially throw an error since it's an impossible/error state. I've had conversations in the past with some folks that think you should _always_ provide a default case no matter what, even if it can never be reached, but I've also heard some disagree with that, so if you strongly prefer to leave it without, then I'm OK with it.